proto/provenance/reward/v1/reward.proto

```diff
@@ -0,0 +1,254 @@
+  // The total amount of all funds claimed by participants for all past claim periods.
+  cosmos.base.v1beta1.Coin claimed_amount = 7 [(gogoproto.nullable) = false];
+  // Maximum reward per claim period per address.
+  cosmos.base.v1beta1.Coin max_reward_by_address = 8 [(gogoproto.nullable) = false];
+  // Minimum amount of coins for a program to rollover.
+  cosmos.base.v1beta1.Coin minimum_rollover_amount = 9 [(gogoproto.nullable) = false];
+  // Number of seconds that a claim period lasts.
+  uint64 claim_period_seconds = 10;
+  // Time that a RewardProgram should start and switch to STARTED state.
+  google.protobuf.Timestamp program_start_time = 11 [
+    (gogoproto.stdtime)  = true,
+    (gogoproto.nullable) = false,
+    (gogoproto.jsontag)  = "program_start_time",
+    (gogoproto.moretags) = "yaml:\"program_start_time\""
+  ];
+  // Time that a RewardProgram is expected to end, based on data when it was setup.
+  google.protobuf.Timestamp expected_program_end_time = 12 [
+    (gogoproto.stdtime)  = true,
+    (gogoproto.nullable) = false,
+    (gogoproto.jsontag)  = "expected_program_end_time",
+    (gogoproto.moretags) = "yaml:\"expected_program_end_time\""
+  ];
+  // Time that a RewardProgram MUST end.
+  google.protobuf.Timestamp program_end_time_max = 13 [
+    (gogoproto.stdtime)  = true,
+    (gogoproto.nullable) = false,
+    (gogoproto.jsontag)  = "program_end_time_max",
+    (gogoproto.moretags) = "yaml:\"program_end_time_max\""
+  ];
+  // Used internally to calculate and track the current claim period's ending time.
+  google.protobuf.Timestamp claim_period_end_time = 14 [
+    (gogoproto.stdtime)  = true,
+    (gogoproto.nullable) = false,
+    (gogoproto.jsontag)  = "claim_period_end_time",
+    (gogoproto.moretags) = "yaml:\"claim_period_end_time\""
+  ];
+  // Time the RewardProgram switched to FINISHED state. Initially set as empty.
+  google.protobuf.Timestamp actual_program_end_time = 15 [
+    (gogoproto.stdtime)  = true,
+    (gogoproto.nullable) = false,
+    (gogoproto.jsontag)  = "actual_program_end_time",
+    (gogoproto.moretags) = "yaml:\"actual_program_end_time\""
+  ];
+  // Number of claim periods this program will run for.
+  uint64 claim_periods = 16;
+  // Current claim period of the RewardProgram. Uses 1-based indexing.
+  uint64 current_claim_period = 17;
+  // maximum number of claim periods a reward program can rollover.
+  uint64 max_rollover_claim_periods = 18;
+  // Current state of the RewardProgram.
+  State state = 19;
+  // Grace period after a RewardProgram FINISHED. It is the number of seconds until a RewardProgram enters the EXPIRED
+  // state.
+  uint64 expiration_offset = 20;
+  // Actions that count towards the reward.
+  repeated QualifyingAction qualifying_actions = 21
+      [(gogoproto.nullable) = false, (gogoproto.moretags) = "yaml:\"qualifying_actions"];
+}
+
+// EventSetRewardProgram event emitted when a reward program is is updated
+message EventSetRewardProgram {
```

Comment:
This is an extensive amount of information in an event... typically speaking events are small and indicate something specific occurred. It would make more sense to indicate the program changed and provide its ID for an interested user to query on than to duplicate the entire state of the object into the event...

Reply:
@Taztingo  i believe deleted it, since it seems a bit too much detail.